homeassistant/components/shelly/cover.py

```diff
@@ -0,0 +1,109 @@
+"""Light for Shelly."""
+from aioshelly import Block
+
+from homeassistant.components.cover import (
+    CoverEntity,
+    SUPPORT_OPEN,
+    SUPPORT_CLOSE,
+    SUPPORT_STOP,
+    SUPPORT_SET_POSITION,
+)
+from homeassistant.const import (
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import callback
+
+from . import ShellyBlockEntity, ShellyDeviceWrapper
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up cover for device."""
+    wrapper = hass.data[DOMAIN][config_entry.entry_id]
+    blocks = [block for block in wrapper.device.blocks if block.type == "roller"]
```

Comment:
Which devices have type = roller? Is that when a Shelly 2.5 is in roller mode? If so, is this check useless? https://github.com/home-assistant/core/blob/dev/homeassistant/components/shelly/switch.py#L15

Reply:
Shelly 2 and 2.5 when in roller mode, so yes check is redundant

Reply:
Actually there are 2 relays in roller mode, so I would keep the check.

```
   "blk":[
        {
            "I":1,
            "D":"relay_0"
        },
        {
            "I":2,
            "D":"relay_1"
        },
        {
            "I":3,
            "D":"roller_0"
        },
        {
            "I":4,
            "D":"device"
        }
    ],
```